<a href="https://colab.research.google.com/github/shimoli007/Git-_Basics/blob/main/5G_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Step 1: Load and Explore the Dataset

a. Let's display the general information about the data

In [ ]:
import pandas as pd

#load the dataset
data = pd.read_csv('/content/5G_energy_consumption_dataset.csv')

#display the general information about the data
data.info()
print("----------")
data.head()
print("----------")
data.tail()
print("---------")
data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92629 entries, 0 to 92628
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Time     92629 non-null  object 
 1   BS       92629 non-null  object 
 2   Energy   92629 non-null  float64
 3   load     92629 non-null  float64
 4   ESMODE   92629 non-null  float64
 5   TXpower  92629 non-null  float64
dtypes: float64(4), object(2)
memory usage: 4.2+ MB
----------
----------
---------


Energy          load        ESMODE       TXpower
count  92629.000000  92629.000000  92629.000000  92629.000000
mean      28.138997      0.244705      0.081361      6.765427
std       13.934645      0.234677      0.382317      0.309929
min        0.747384      0.000000      0.000000      5.381166
25%       18.236173      0.057370      0.000000      6.427504
50%       24.065770      0.165550      0.000000      6.875934
75%       35.724963      0.363766      0.000000      6.875934
max      100.000000      0.993957      4.000000      8.375336

The data displayed above contains 92629 entries.
It has also has 6 columns:

*   Time
*   BS
*   Energy
*   Load
*   ESMODE
*   TXpower

b. Let's create the pandas profiling report for better insights

In [ ]:
!pip install ydata-profiling
from ydata_profiling import ProfileReport
profile = ProfileReport(data)
profile.to_file(output_file='report.html')

  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 356.2/356.2 kB 13.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 296.5/296.5 kB 18.6 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 686.1/686.1 kB 32.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 104.8/104.8 kB 6.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.5/4.5 MB 64.3 MB/s eta 0:00:00
  Created wheel for htmlmin: filename=htmlmin-0.1.12-py3-none-any.whl size=27081 sha256=52b03495ff784e0c75bbbf6fb68f5680862a80dfb65fd78e6422aae94c74638d
  Stored in directory: /root/.cache/pip/wheels/dd/91/29/a79cecb328d01739e64017b6fb9a1ab9d8cb1853098ec5966d
Successfully built htmlmin


/usr/local/lib/python3.10/dist-packages/ydata_profiling/profile_report.py:363: UserWarning: Try running command: 'pip install --upgrade Pillow' to avoid ValueError
  warnings.warn(


Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

c. Handle the missing or corrupted data

In [ ]:
#separate numeric and non-numeric columns
numeric_columns = data.select_dtypes(include=['number']).columns
non_numeric_columns = data.select_dtypes(exclude=['number']).columns

#fill missing values in numeric columns with mean
data[numeric_columns] = data[numeric_columns].fillna(data[numeric_columns].mean())

#fill missing values in non-numeric columns with mode
data[non_numeric_columns] = data[non_numeric_columns].fillna(data[non_numeric_columns].mode().iloc[0])

#check again for missing values
print(data.isnull().sum())

Time       0
BS         0
Energy     0
load       0
ESMODE     0
TXpower    0
dtype: int64
